### PR TITLE
fix(ext/node): basic vm.runInNewContext implementation

### PR DESCRIPTION
**Pull request by littledivy**
Simple implementation to support webpack (& Next.js): https://github.com/webpack/webpack/blob/87660921808566ef3b8796f8df61bd79fc026108/lib/javascript/JavascriptParser.js#L4329

### CODE REVIEW

File: ext/node/polyfills/vm.ts [hunk 1]
```diff
@@ -25,8 +26,8 @@ export class Script {
     notImplemented("Script.prototype.runInContext");
   }
 
-  runInNewContext(_contextObject: any, _options: any) {
-    notImplemented("Script.prototype.runInNewContext");
+  runInNewContext(contextObject: any, _options: any) {
```
Comment (bartlomieju) [inline ext/node/polyfills/vm.ts:29]: Change to `options` and if it's not undefined print a warning that this option is currently not supported.

File: ext/node/polyfills/vm.ts [hunk 2]
```diff
@@ -51,11 +52,11 @@ export function runInContext(
 }
 
 export function runInNewContext(
-  _code: string,
-  _contextObject: any,
+  code: string,
+  contextObject: any,
   _options: any,
```
Comment (bartlomieju) [inline ext/node/polyfills/vm.ts:57]: Ditto.

**Review by bartlomieju — COMMENTED** [review]
Please add one or two test cases. Try to evaluate any code and then try to evaluate a code that tries to access something from the global context (and throws).

**Comment by kt3k** [comment]
When the given script throws, then it seems panicking. Can this be avoided somehow?

```
$ cat a.mjs 
import { runInNewContext } from "node:vm";
runInNewContext("throw new Error('abc')");
$ ./target/debug/deno run a.mjs

============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos aarch64
Version: 1.38.5
Args: ["./target/debug/deno", "run", "a.mjs"]

thread 'main' panicked at ext/node/ops/v8.rs:53:34:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by littledivy** [comment]
Just fixed that @kt3k :)

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**